cmd/thanos/compact.go

```diff
@@ -284,9 +284,35 @@ func runCompact(
 			if err == nil {
 				return nil
 			}
+
+			if multiErr, ok := err.(tsdb.MultiError); ok {
```

Comment:
So this to me looks like a bit of duplication. Can we make sure `IsHaltError` checks if error is multierror and then checks if any has halt? (: And same for retry?

Reply:
@bwplotka great idea, I've updated the error checking functions to handle multierrors 👍 Hopefully this is good to go now.